client/src/main/scala/org/apache/celeborn/client/WorkerStatusTracker.scala

```diff
@@ -188,9 +213,43 @@ class WorkerStatusTracker(
           statusChanged = true
         }
       }
-      val retainResult = shuttingWorkers.retainAll(res.shuttingWorkers)
-      val addResult = shuttingWorkers.addAll(res.shuttingWorkers)
-      statusChanged = statusChanged || retainResult || addResult
+
+      val retainShuttingWorkersResult = shuttingWorkers.retainAll(res.shuttingWorkers)
+      val addShuttingWorkersResult = shuttingWorkers.addAll(res.shuttingWorkers)
+
+      if (changPartitionWithAvailableWorkers) {
+        // AvailableWorkers filter Client excludedWorkers and shuttingWorkers.
+        // AvailableWorkers already filtered res.excludedWorkers and res.shuttingWorkers.
+        val resAvailableWorkers: JSet[WorkerInfo] = new JHashSet[WorkerInfo](res.availableWorkers)
```

Comment:
remove excludedWorkers/shuttingWorkers update logic which have already processed before.

Reply:
Since availableWorkers filtering excludedWorkers/shuttingWorkers in heartBeat cannot cover the condition that new excludedWorkers being added between two heartBeats interval , we have to filter excludedWorkers everyTime we use the availableWorkers~